### ISSUE TITLE
evaluate libp2p as backend network stack

### ISSUE THREAD

**Issue by martinjaggi** [epfml/disco#3]
https://libp2p.io/

evaluate which bindings or deployments of libp2p would be most suitable for communicating gradients as in ML training, supporting mobile phone OSs, and for interfacing with e.g. PyTorch / jax or similar schemes

**Comment by martinjaggi** [comment]
not an urgency now because we have the peer.js version,
but we should come back to this issue sometime later, as libp2p offers many nice advantages such as elegant server-free peer discovery (using DHTs). we could use that later to discover peers who want to train the same task, similarly as the keyword based approach used here:  https://github.com/dennis-tra/pcp

**Comment by martinjaggi** [comment]
we should keep an eye on when the js version of libp2p would be ready to offer the (basic) functionality we need: https://github.com/libp2p/js-libp2p
